src/utils/appLinking.js

```diff
@@ -0,0 +1,119 @@
+import { Linking } from "react-native";
+import { Alert } from "../components/general/Alert";
+
+/**
+ * @description Object that holds the slack team id and several channel ids.
+ * The information in here is the most current as of May 28th, 2020.
+ * To find the team id or channel id login in slack on a computer.
+ * @typedef {{team: String, channel: String, fbGroupshpeucf: String}} slackInfo
+ * @const {slackInfo} slackInfo
+ */
+
+export const slackInfo = {
+
+	team: "TC61JSPUZ",
+	channel: {
+		general: "CPRDXHHD4",
+		announcements: "CC5S86GHE",
+		fundraisingCommitte: "CCUUWRU1Y",
+		motorshpe: "CP5HC76TD",
+		projects: "CNATZRD9C",
+		shpejr: "GCWFE5630"
+	},
+	fbGroupshpeucf: "120691161371846"
+};
+
+/**
+ * @description A  shorthand function that returns the uri scheme and universal for opening an app and pass parameters to navigate
+ * in the app if parameters are included.
+ * @param {String}    inquiry  The value of the query.
+ * @param {String=}   intent   The vlalue that specify the type of query.
+ * @example Call the function like this formatUrl[name of the app](inquiry, intent)
+ */
+
+const formatUrl = {
+	email: (inquiry) => (inquiry) = ({ uri: `mailto:${inquiry}`, url: `mailto:${inquiry}` }),
+
+	facebook: (inquiry, intent) => (intent == "groups" && inquiry) ? ({ uri: `https://facebook.com/groups/${inquiry}`, url: `https://www.facebook.com/groups/${inquiry}/` }) :
+		(intent == "profile" && inquiry) ? ({ uri: `https://facebook.com/${inquiry}`, url: `https://www.facebook.com/${inquiry}` }) : ({ uri: "fb://", url: "https://www.facebook.com/" }),
+
+	instagram: (inquiry) => (inquiry) ? ({ uri: `instagram://user?username=${inquiry}`, url: `https://www.instagram.com/_u/${inquiry}/` }) :
+		({ uri: "instagram://", url: "https://www.instagram.com" }),
+
+	linkedin: (inquiry, intent) => (intent === "profile" && inquiry) ? ({ uri: `linkedin://in/${inquiry}`, url: `https://www.linkedin.com/in/${inquiry}` }) :
+		(intent === "company" && inquiry) ? ({ uri: `linkedin://company/${inquiry}`, url: `https://www.linkedin.com/company/${inquiry}` }) :
+			({ uri: "linkedin://", url: "https://www.linkedin.com" }),
+
+	slack: (inquiry, intent) => (intent === "channel" && (inquiry = inquiry in slackInfo.channel ? slackInfo.channel[inquiry] : inquiry)) ?
+		({ uri: `slack://channel?team=${slackInfo.team}&id=${inquiry}` }) : ({ uri: `slack://open?team=${slackInfo.team}` }),
+
+	phone: (inquiry) => (inquiry) = ({ uri: `tel:+${inquiry}`, url: `tel:+${inquiry}` }),
+
+	web: (inquiry) => (inquiry) = ({ uri: `${inquiry}`, url: `${inquiry}` })
+};
```

Comment:
This is a little hard to read because there are so many objects all put together. I think if you separate things into multiple lines it might help., Not 100% sure about that. If you separate it and don't think it helps then you can put it back.